crates/rooch/src/commands/account/commands/sign.rs

```diff
@@ -1,61 +1,120 @@
 // Copyright (c) RoochNetwork
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::cli_types::{CommandAction, WalletContextOptions};
+use crate::cli_types::{CommandAction, TransactionOptions, WalletContextOptions};
+use anyhow::Result;
 use async_trait::async_trait;
 use clap::Parser;
-use hex::ToHex;
+use move_command_line_common::types::ParsedStructType;
+use move_core_types::language_storage::TypeTag;
+use moveos_types::{state::MoveState, transaction::MoveAction};
 use rooch_key::keystore::account_keystore::AccountKeystore;
-use rooch_rpc_api::jsonrpc_types::account_sign_view::AccountSignView;
 use rooch_types::{
-    address::ParsedAddress,
-    error::{RoochError, RoochResult},
+    error::RoochResult,
+    framework::auth_payload::{AuthPayload, SignData},
+    function_arg::{parse_function_arg, FunctionArg, ParsedFunctionId},
 };
+use serde::{Deserialize, Serialize};
 
-/// Sign an msg with current account private key (sign_hashed)
-///
-/// This operation must be specified with -a or
-/// --address to export only one address with a private key.
+/// Sign a tx with an account
 #[derive(Debug, Parser)]
 pub struct SignCommand {
-    #[clap(short = 'a', long = "address", value_parser=ParsedAddress::parse, default_value = "")]
-    address: ParsedAddress,
+    /// Function name as `<ADDRESS>::<MODULE_ID>::<FUNCTION_NAME>`
+    /// Example: `0x42::message::set_message`, `rooch_framework::empty::empty`
+    #[clap(long)]
+    pub function: ParsedFunctionId,
+
+    /// TypeTag arguments separated by spaces.
+    ///
+    /// Example: `0x1::M::T1 0x1::M::T2 rooch_framework::empty::Empty`
+    #[clap(
+        long = "type-args",
+        value_parser=ParsedStructType::parse,
+    )]
+    pub type_args: Vec<ParsedStructType>,
+
+    /// Arguments combined with their type separated by spaces.
+    ///
+    /// Supported types [u8, u16, u32, u64, u128, u256, bool, object_id, string, address, vector<inner_type>]
+    ///
+    /// Example: `address:0x1 bool:true u8:0 u256:1234 'vector<u32>:a,b,c,d'`
+    ///     address and uint can be written in short form like `@0x1 1u8 4123u256`.
+    #[clap(long = "args", value_parser=parse_function_arg)]
+    pub args: Vec<FunctionArg>,
+
+    #[clap(flatten)]
+    pub tx_options: TransactionOptions,
```

Comment:
We need a sign message command, not sign tx. the `tx sign` is for sign tx.

Reply:
> We need a sign message command, not sign tx. the `tx sign` is for sign tx.

Noted.